### PR TITLE
Allow nested entities to be interaction targets

### PR DESCRIPTION
**Pull request by 0x0539**
Fixes an issue discussed in the Discord where nested entities don't work correctly as interaction targets. This was due to the code using Transform instead of GlobalTransform.  I tested that Follower interaction still works after this change.

### CODE REVIEW

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Thanks!
Two little things left:
- Remove the .swp file
- Make the system run after [TransformPropagate](https://docs.rs/bevy/latest/bevy/transform/enum.TransformSystem.html#variant.TransformPropagate).

**Comment by 0x0539** [comment]
Good catches, thanks!

**Comment by janhohenheim** [comment]
Looks good now, merging after CI :)
Thanks for your work!

**Review by janhohenheim — APPROVED** [review]

File: src/world_interaction/interactions_ui.rs [hunk 1]
```diff
@@ -9,7 +9,11 @@ use crate::{
 
 use crate::{world_interaction::dialog::DialogTarget, GameState};
 use anyhow::Context;
-use bevy::{prelude::*, window::PrimaryWindow};
+use bevy::{
+    prelude::*, 
+    transform::TransformSystem::TransformPropagate
```
Comment (janhohenheim) [inline src/world_interaction/interactions_ui.rs:14]:
```suggestion
    transform::TransformSystem::TransformPropagate,
```

**Review by janhohenheim — CHANGES_REQUESTED** [review]

**Comment by 0x0539** [comment]
Yeah... Hey at least I know how to run the linter now

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**